doc/BUILD.md

```diff
@@ -1,80 +1,109 @@
-* To build **Navigation2** package for specific distro branch (e.g- dashing-devel), select the matching ROS2 distro in place of `ros2-distro-name` during `rosdep install`.
+---
 
+## Build Options
 
-### 2.1 ROS2
-* Install ROS2 with dependencies from the ROS2 Installation Page: https://index.ros.org/doc/ros2/Installation/
-* Install the latest distribution of ROS2 to support Navigation2 `master`.
-* ROS2 can be installed either from binary packages or build from source. Choose to build ROS2 from source to support Navigation2 `master`.
+There two different recommended methods for building Navigation2 stack from source.
 
-### 2.2 Navigation2 Dependencies
-If ROS2 installed from source (Ignore if installed from binary packages)
-```console
-$ source ~/ros2_ws/install/setup.bash
-```
-Fetch, build and install navigation2 dependencies:
-```console
-$ mkdir -p ~/ros2_nav_dependencies_ws/src
-$ cd ~/ros2_nav_dependencies_ws
-$ wget https://raw.githubusercontent.com/ros-planning/navigation2/master/tools/ros2_dependencies.repos
-$ vcs import src < ros2_dependencies.repos
-$ rosdep install -y -r -q --from-paths src --ignore-src --rosdistro <ros2-distro-name>
-e.g- ros2-distro-name = dashing
-$ colcon build --symlink-install
-```
+### 1- Easy Build
+* An automated method to build Navigation2 in a simple and easy way.
+* To build Navigation2 by using the build script, go to [QuickStart](#2-quickstart-using-initial-setup-script)
+
+---
 
-### 2.3 Navigation2
-
-Fetch, build and install navigation2 stack:
-```console
-$ source ~/ros2_nav_dependencies_ws/install/setup.bash
-$ mkdir -p ~/navigation2_ws/src
-$ cd ~/navigation2_ws/src
-$ git clone https://github.com/ros-planning/navigation2.git -b <nav2-branch-name>
-e.g- nav2-branch-name = master
-$ cd ~/navigation2_ws
-$ rosdep install -y -r -q --from-paths src --ignore-src --rosdistro <ros2-distro-name>
-e.g- ros2-distro-name = dashing
-$ colcon build --symlink-install
+### 2- Manually Build from Source
+
+#### Step 1- Build ROS 2
+* [ROS2 Build Instructions](https://index.ros.org/doc/ros2/Installation/)
+  * When building ROS 2 from source, if you want to get the latest version of the ROS 2 packages, make sure that the `ros2.repos` file is from the `master` branch.
+  * When installing the ROS 2 dependencies using rosdep, make sure to set `--rosdistro` to the lastest ROS 2 distribution (e.g `eloquent`)
+
+#### Step 2- Build Navigation2 Dependencies
+
+* If you built ROS 2 from source, source the setup.bash file in the ROS 2 build workspace.
+    ```sh
+  source ~/ros2_ws/install/setup.bash
+  ```
+* If you installed ROS 2 from the binary packages, source the setup.bash file in the ROS 2 installation directory.
+  ```sh 
+  source /opt/ros/<ros2-distro>/setup.bash
+  ```
+
+##### Build and install navigation2 dependencies:
+
+```sh
+mkdir -p ~/nav2_depend_ws/src
+cd ~/nav2_depend_ws
+wget https://raw.githubusercontent.com/ros-planning/navigation2/master/tools/ros2_dependencies.repos
```

Comment:
I'd rather poeple checkout out navigation 2 into the nav2 workspace first, then did a vcs import src < nav2_ws/navigation2/tools/ros2_dependencies.

That works no matter which version of the nav stack they are working with.

Reply:
I agree but in my opinion, this way it is simpler to follow. Otherwise, I need to add one or two more steps for creating workspaces and cloning/downloading repositories.